### PR TITLE
Establish basis for examples

### PR DESCRIPTION
**Pull request by BarbourSmith**
The pyGithub documentation is excellent, but interacting with the Github api for the first time can be confusing without any examples.

There have been a number of requests for more examples of how to use PyGithub.

This pull request establishes an examples.rst file which serves as a place for the community to contribute examples.

The examples file starts off with some from @pconrad showing off the basic use of PyGithub and some from me showing how to interact with pull requests as issues.

This pull request is an attempt to begin to address issue #321

### CODE REVIEW

**Comment by erinallard-infoscout** [comment]
Hi @sfdye - would love to have your feedback on these first examples. What do you think?

File: doc/examples.rst [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+.. codeblock:: python
+    from github import Github
+    import os
+
+    GITHUB_API_KEY = os.environ['GITHUB_API_KEY']
+    g = Github(GITHUB_API_KEY)
+        
+In the code snippet above, we import the ``Github`` class from pyGitHub and also import Python's ``os`` package. The ``os`` package gives us a way to pull in the GitHub API key that the computer can now see after sourcing ``secrets.sh``. We then pass the API key in as an argument when creating a new ``Github`` instance called ``g``. From here, we can call `any of the methods that are associated with the Github object <http://pygithub.readthedocs.io/en/latest/github.html>`_ on **our** instance, ``g``. For example, we can call `the get_user() method <http://pygithub.readthedocs.io/en/latest/github.html#github.MainClass.Github.get_user>`_ which returns a `NamedUser object <http://pygithub.readthedocs.io/en/latest/github_objects/NamedUser.html#github.NamedUser.NamedUser>`_, which itself gives us access to attributes like the user's avatar URL, bio, company, email, etc.
+
+.. codeblock:: python
```
Comment (sfdye) [inline doc/examples.rst]: Looks like this code snippet is not being properly formatted.

File: doc/examples.rst [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+    ``export GITHUB_API_KEY='yourkeygoeshere'``
+    
+    If you run ``git status`` at this point you'll see that ``secrets.sh`` does not appear in the output. It's safely hiding inside ``.gitignore``.
+    
+(3) In your Terminal -- and you're inside an environment, right? -- run ``source secrets.sh``. This will make your GitHub API key visible to your computer.
+
+(4) Next, create a new file in your project directory that will hold your code. In this example we'll call it ``try_pygithub.py``. Copy and paste this code into ``try_pygithub.py``. Be sure to read the explanation that follows.
+
+.. codeblock:: python
```
Comment (sfdye) [inline doc/examples.rst]: Same here

**Comment by sfdye** [comment]
Thanks very much for putting these together. As a general guideline, the examples should be generic enough to accommodate a variety of audiences, and should also conform to some standards. (like variable naming convention and coding style, etc). I suggest we can follow the Python community practice by using `some_var` instead of `someVar`, and follow PEP8 for coding style.

**Comment by sfdye** [comment]
If you are looking for some inspiration, boto from aws is a good reference.

Read the docs: https://boto3.readthedocs.io/en/latest/guide/examples.html
Source: https://github.com/boto/boto3/tree/develop/docs/source/guide

**Comment by sfdye** [comment]
Sorry stale bot had to close this PR. Now with #874 , we can probably port some of the examples over. What do you think? @BarbourSmith

**Comment by BarbourSmith** [comment]
Yes! No worries on the stale bot, we do the same things on our projects. 

How do we feel about me making separate pull requests for specific examples once #875 has created the place for them. I would love a second set of eyes on each example I propose because I'm a little rusty now

**Pull request closed without merging**